src/main/java/com/revature/studyforce/notification/service/FlashcardSubscriptionService.java

```diff
@@ -0,0 +1,130 @@
+     */
+    public FlashcardSubscription createFlashcardSubscription(Integer flashcardId, Integer userId){
+        return flashcardSubscriptionRepository.save(
+            new FlashcardSubscription(
+                    Objects.requireNonNull(flashcardRepository.findById(flashcardId).orElse(null)),
+                    subscriptionService.getSubscriptionByUserId(userId)
+            )
+        );
+    }
+
+    /**
+     * Deletes a flashcard subscription for a given flashcard and user subscription {@link FlashcardSubscriptionRepository#delete(Object)}
+     * The subscription is built by grabbing the flashcard from {@link FlashcardRepository#findById(Object)}
+     * And the user subscription from {@link SubscriptionService#getSubscriptionByUserId(Integer)}
+     * @param flashcardId The flashcardId of the flashcard to be unsubscribed from
+     * @param userId The userId of the user to be unsubscribed
+     * @return The deleted flashcard subscription.
+     */
+    public FlashcardSubscription deleteFlashcardSubscription(Integer flashcardId, Integer userId){
+        FlashcardSubscription subscription = new FlashcardSubscription(
+            Objects.requireNonNull(flashcardRepository.findById(flashcardId).orElse(null)),
+            subscriptionService.getSubscriptionByUserId(userId)
+        );
+        flashcardSubscriptionRepository.delete(subscription);
+        return subscription;
+    }
+
+    /**
+     * Deletes a list of subscriptions {@link FlashcardSubscriptionRepository#deleteAll(Iterable)}
+     * @param subscriptions The list of subscriptions to be deleted.
+     * @return The list of deleted subscriptions.
+     */
+    public List<FlashcardSubscription> deleteAllFlashcardSubscriptions(List<FlashcardSubscription> subscriptions){
+        flashcardSubscriptionRepository.deleteAll(subscriptions);
+        return subscriptions;
+    }
+
+    /**
+     * Generates a list of subscriptions for a given flashcard to be deleted by /deleteAllFlashcardSubscriptions
+     * @param flashcardId The flashcardId of the flashcard to delete all subscribers.
+     * @return The list of deleted subscriptions.
+     */
+    public List<FlashcardSubscription> deleteAllFlashcardSubscriptionsByFlashcardId(Integer flashcardId){
+        List<FlashcardSubscription> subscriptions = getAllSubscribersByFlashcardId(flashcardId);
+        return deleteAllFlashcardSubscriptions(subscriptions);
+    }
+
+    /**
+     * Generates a list of subscriptions for a given user to be deleted by /deleteAllFlashcardSubscriptions
+     * @param userId The userId of the user to delete all subscriptions.
+     * @return The list of deleted subscriptions.
+     */
+    public List<FlashcardSubscription> deleteAllFlashcardSubscriptionsByUserId(Integer userId){
+        List<FlashcardSubscription> subscriptions = getAllSubscriptionsByUserId(userId);
+        return deleteAllFlashcardSubscriptions(subscriptions);
+    }
+}
```

Comment:
Refactor with pagination

Reply:
This is an internal service method, not something to be hit by an endpoint and served in the frontend.